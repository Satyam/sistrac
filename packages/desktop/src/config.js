module.exports = {
  REST_PORT: 8080,
  REST_HOST: 'http://localhost',
  REST_API_PATH: '/data/v2',
  SESSION_TIMEOUT: 20 * 60 * 1000,
  ROOT_DIR: __dirname,
  SQL_HOST: 'localhost',
  DB_NAME: 'sistrac',
  SQL_USER: 'root',
  SQL_PASSWORD: '',
  SECRET: 'Sistrac clave secreta: pepino',
  COOKIE_NAME: 'user',
  APP_HOST: 'http://localhost',
  APP_PORT: '3000',
};
