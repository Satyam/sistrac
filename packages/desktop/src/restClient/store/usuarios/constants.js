export const NAME = 'usuarios';

export const GET_USUARIOS = `${NAME} / get usuarios`;

export const GET_USUARIO = `${NAME} / get usuario`;
export const UPDATE_USUARIO = `${NAME} / update usuario`;
export const CREATE_USUARIO = `${NAME} / create usuario`;
export const DELETE_USUARIO = `${NAME} / delete usuario`;

export const LOGIN = `${NAME} / login`;
export const GET_USUARIO_ACTUAL = `${NAME} / get usuario actual`;
export const LOGOUT = `${NAME} / logout`;

export const STATUS_INITIAL = 0;
export const STATUS_UNAUTHORIZED = 1;
export const STATUS_LOGGED_IN = 2;
export const STATUS_GETTING_CURRENT_USER = 3;
export const STATUS_LOGGED_OUT = 4;
