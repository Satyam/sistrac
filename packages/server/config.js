module.exports = Object.assign(require('../../config.js'), {});

