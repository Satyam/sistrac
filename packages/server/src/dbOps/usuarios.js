let db;

export default function init(db1) {
  db = db1;
}

function convertToBoolean(row) {
  return (
    (row || false) && {
      ...row,
      rolGuarda: !!row.rolGuarda,
      rolDios: !!row.rolDios,
      rolSupervisor: !!row.rolSupervisor,
      rolMecanico: !!row.rolMecanico,
    }
  );
}
export async function readUsuarios(idUsuarios) {
  const where = idUsuarios ? 'where idUsuario in (?)' : '';
  const query = await db.query(`select * from UsuariosSimple ${where}`, [
    idUsuarios,
  ]);
  return query.map(convertToBoolean);
}

export async function createUsuario(values) {
  try {
    const query = await db.query('insert into Usuarios set ?', values);
    return query.insertId;
  } catch (err) {
    if (err.code === 'ER_DUP_ENTRY') {
      return false;
    }
    throw err;
  }
}

export async function readUsuario(id) {
  return convertToBoolean(
    await db.queryRow('select * from UsuariosSimple where idUsuario = ?', [id]),
  );
}

export async function readUsuarioPorUsuario(usuario) {
  return convertToBoolean(
    await db.queryRow('select * from UsuariosSimple where usuario = ?', [
      usuario,
    ]),
  );
}

export async function loginUsuario(usuario, password) {
  return convertToBoolean(
    await db.queryRow(
      `select idUsuario, usuario, nombre, nivel, 
    rolDios, rolGuarda, rolMecanico, rolSupervisor, funcion 
    from Usuarios where usuario = ? and password = ?`,
      [usuario, password],
    ),
  );
}

export async function deleteUsuario(id) {
  const query = await db.query('delete from Usuarios where idUsuario = ?', [
    id,
  ]);
  return query.affectedRows === 1;
}

export async function updateUsuario(id, datos) {
  const query = await db.query('update Usuarios set ? where idUsuario = ?', [
    datos,
    id,
  ]);
  return query.affectedRows === 1;
}
