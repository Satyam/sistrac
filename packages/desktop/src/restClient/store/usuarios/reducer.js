import indexBy from '_store/utils/indexBy';

import {
  REPLY_RECEIVED,
  REQUEST_SENT,
  FAILURE_RECEIVED,
} from '_store/utils/promiseMiddleware';

import {
  // NAME,
  GET_USUARIOS,
  GET_USUARIO,
  // UPDATE_USUARIO,
  // CREATE_USUARIO,
  // DELETE_USUARIO,
  LOGIN,
  LOGOUT,
  GET_USUARIO_ACTUAL,
} from './constants';

import { SESSION_TIMEOUT } from '_src/config';

export const STATUS_INITIAL = 0;
export const STATUS_UNAUTHORIZED = 1;
export const STATUS_LOGGED_IN = 2;
export const STATUS_GETTING_CURRENT_USER = 3;
export const STATUS_LOGGED_OUT = 4;

export default (
  state = { hash: {}, activo: null, vence: null, status: STATUS_INITIAL },
  action,
) => {
  const stage = action && action.stage;
  switch (stage) {
    case FAILURE_RECEIVED:
      if (action.error === 401) {
        // 401 HTTP code: Unauthorized
        return {
          ...state,
          activo: null,
          vence: null,
          status:
            state.prevStatus === STATUS_INITIAL
              ? STATUS_LOGGED_OUT
              : STATUS_UNAUTHORIZED,
          prevStatus: null,
        };
      }
      return state;
    case REQUEST_SENT:
      switch (action.type) {
        case GET_USUARIO_ACTUAL:
          return {
            ...state,
            status: STATUS_GETTING_CURRENT_USER,
            prevStatus: state.status,
          };
        case GET_USUARIOS: {
          const idUsuarios = action.payload.faltantes;
          if (!idUsuarios) return state;
          return {
            ...state,
            hash: idUsuarios.reduce(
              (hash, idUsuario) => ({
                ...hash,
                [idUsuario]: { idUsuario },
              }),
              state.hash,
            ),
          };
        }
        default:
          return state;
      }
    case REPLY_RECEIVED:
      switch (action.type) {
        case GET_USUARIO_ACTUAL:
        case LOGIN: {
          const usuario = action.payload;
          const idUsuario = usuario.idUsuario;
          return {
            ...state,
            activo: idUsuario,
            vence: Date.now() + SESSION_TIMEOUT * 1000,
            hash: { ...state.hash, [idUsuario]: usuario },
            status: STATUS_LOGGED_IN,
          };
        }
        case LOGOUT:
          return {
            ...state,
            activo: null,
            vence: null,
            status: STATUS_LOGGED_OUT,
          };
        case GET_USUARIO: {
          const usuario = action.payload;
          const idUsuario = usuario.idUsuario;
          return {
            ...state,
            hash: { ...state.hash, [idUsuario]: usuario },
          };
        }
        case GET_USUARIOS:
          return {
            ...state,
            hash: indexBy(action.payload, 'idUsuario', state.hash),
          };
        default:
          return state;
      }
    default:
      return state;
  }
};
