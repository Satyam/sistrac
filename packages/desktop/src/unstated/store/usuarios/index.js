import { Container } from 'unstated';
import restAPI from '_store/utils/restAPI';
import indexBy from '_store/utils/indexBy';

import { SESSION_TIMEOUT } from '_src/config';

import {
  NAME,
  STATUS_INITIAL,
  STATUS_UNAUTHORIZED,
  STATUS_LOGGED_IN,
  STATUS_GETTING_CURRENT_USER,
  STATUS_LOGGED_OUT,
} from './constants';

const api = restAPI(NAME);

export default class Usuarios extends Container {
  // these are mostly like the reducers
  state = { hash: {}, activo: null, vence: null, status: STATUS_INITIAL };
  failureReceived = err => {
    if (err.code === 401) {
      this.noUsuarioActual();
    } else throw err;
  };
  noUsuarioActual() {
    this.setState({
      activo: null,
      vence: null,
      status:
        this.state.prevStatus === STATUS_INITIAL
          ? STATUS_LOGGED_OUT
          : STATUS_UNAUTHORIZED,
      prevStatus: null,
    });
  }
  usuarioActualRequested = () => {
    this.setState({
      status: STATUS_GETTING_CURRENT_USER,
      prevStatus: this.state.status,
    });
  };
  usuariosRequested = idUsuarios => {
    if (!idUsuarios) return;
    this.setState({
      hash: idUsuarios.reduce(
        (hash, idUsuario) => ({
          ...hash,
          [idUsuario]: { idUsuario },
        }),
        this.state.hash,
      ),
    });
  };
  usuariosRead = usuarios => {
    this.setState({
      hash: indexBy(usuarios, 'idUsuario', this.state.hash),
    });
    return usuarios;
  };
  loggedIn = usuario => {
    const { idUsuario } = usuario;
    this.setState({
      activo: idUsuario,
      vence: Date.now() + SESSION_TIMEOUT * 1000,
      hash: { ...this.state.hash, [idUsuario]: usuario },
      status: STATUS_LOGGED_IN,
    });
    return usuario;
  };
  loggedOut = () => {
    this.setState({
      activo: null,
      vence: null,
      status: STATUS_LOGGED_OUT,
    });
  };

  // these are mostly like actions:
  login(usuario, password) {
    return api
      .update('/login', { usuario, password })
      .then(this.loggedIn)
      .catch(this.failureReceived);
  }
  logout() {
    return api
      .read('/logout')
      .then(this.loggedOut)
      .catch(this.failureReceived);
  }
  getUsuarioActual() {
    this.usuarioActualRequested();
    return api
      .read('/__actual')
      .then(this.loggedIn)
      .catch(err => {
        if (err.code === 404 || err.code === 401) {
          this.noUsuarioActual();
        } else {
          throw err;
        }
      });
  }
  getUsuarios(usuarios) {
    let faltantes = [];
    if (usuarios) {
      faltantes = usuarios.filter(idUsuario => !this.state.hash[idUsuario]);
      if (!faltantes.length) return;
      this.usuariosRequested(faltantes);
    }
    return api
      .read(`/${faltantes.join(',')}`)
      .then(this.usuariosRead)
      .catch(this.failureReceived);
  }

  // and these, the selectors:
  selUsuarioActivo() {
    const { activo, vence, hash } = this.state;

    if (vence < Date.now()) return {};
    if (!activo) return {};
    return hash[activo];
  }
  selStatusUsuario() {
    return this.state.status;
  }
  selUsuario(idUsuario) {
    return this.state.hash[idUsuario];
  }
}
