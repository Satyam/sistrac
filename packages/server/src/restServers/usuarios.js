import jwt from 'jsonwebtoken';
import md5 from 'md5';
import join from './plainJoin';

import { SECRET, SESSION_TIMEOUT, COOKIE_NAME } from '../config';

import {
  readUsuarioPorUsuario,
  createUsuario,
  deleteUsuario,
  updateUsuario,
  readUsuarios,
} from '../dbOps/usuarios';

import {
  CREATED,
  NOT_FOUND,
  NO_CONTENT,
  UNAUTHORIZED,
  BAD_REQUEST,
  CONFLICT,
} from './httpStatusCodes';

export function setCookie(res, data) {
  res.cookie(COOKIE_NAME, jwt.sign(data, SECRET), {
    maxAge: SESSION_TIMEOUT,
  });
}

const bypassAuthenticationPattern = /\/((login)|(signup))$/;

export function authenticate(req, res, next) {
  if (req.method === 'OPTIONS' || bypassAuthenticationPattern.test(req.path)) {
    next();
    return;
  }
  const cookie = req.cookies && req.cookies[COOKIE_NAME];
  if (cookie) {
    try {
      const { iat, ...user } = jwt.verify(cookie, SECRET);
      setCookie(res, user);
      req.user = user;
      next();
      return;
    } catch (err) {
      res.status(UNAUTHORIZED).json(err);
    }
  }
  res.status(UNAUTHORIZED).end();
}

export default async function(dataRouter, path) {
  dataRouter.put(join(path, '/login'), async (req, res) => {
    const { password, ...user } = await readUsuarioPorUsuario(req.body.usuario);
    if (Object.keys(user).length) {
      const isMatch = md5(req.body.password) === password;
      if (isMatch) {
        setCookie(res, user);
        res.json(user);
        return;
      }
    }
    res.clearCookie(COOKIE_NAME);
    res.status(UNAUTHORIZED).end();
  });

  dataRouter.post(join(path, '/signup'), async (req, res) => {
    const { usuario, password, nombre } = req.body;
    res.clearCookie(COOKIE_NAME);
    if (!usuario || !password || !nombre) {
      res
        .status(BAD_REQUEST)
        .send(
          'Favor de indicar código de usuario, contraseña y nombre completo.',
        );
    } else {
      const resp = await createUsuario({
        ...req.body,
        password: md5(password),
      });
      res.status(resp ? CREATED : CONFLICT).end();
    }
  });

  dataRouter.get(join(path, '/logout'), (req, res) => {
    res.clearCookie(COOKIE_NAME);
    res.status(NO_CONTENT).end();
  });

  dataRouter.get(join(path, '/_actual'), async (req, res) => {
    const { password, ...user } = await readUsuarioPorUsuario(req.user.usuario);
    if (Object.keys(user).length) res.json(user);
    else res.status(NOT_FOUND).end();
  });

  dataRouter.get(join(path, '/:usuario'), async (req, res) => {
    const { usuario } = req.params;
    const list = usuario.split(',');
    if (list.length > 1) {
      res.json(await readUsuarios(list));
    } else {
      const { password, ...user } = await readUsuarioPorUsuario(usuario);
      if (Object.keys(user).length) res.json(user);
      else res.status(NOT_FOUND).end();
    }
  });

  dataRouter.delete(join(path, '/:idUsuario'), async (req, res) => {
    const resp = await deleteUsuario(parseInt(req.params.idUsuario, 10));
    res.status(resp ? NO_CONTENT : NOT_FOUND).end();
  });

  dataRouter.put(join(path, '/:idUsuario'), async (req, res) => {
    const resp = await updateUsuario(
      parseInt(req.params.idUsuario, 10),
      req.body,
    );
    res.status(resp ? NO_CONTENT : NOT_FOUND).end();
  });
}

export function esDios(req, res, next) {
  if (req.user && req.user.rolDios) {
    next();
  } else {
    res.status(UNAUTHORIZED).end();
  }
}

export function esGuarda(req, res, next) {
  if (req.user && req.user.rolGuarda) {
    next();
  } else {
    res.status(UNAUTHORIZED).end();
  }
}

export function esMecanico(req, res, next) {
  if (req.user && req.user.rolMecanico) {
    next();
  } else {
    res.status(UNAUTHORIZED).end();
  }
}

export function esSupervisor(req, res, next) {
  if (req.user && req.user.rolSupervisor) {
    next();
  } else {
    res.status(UNAUTHORIZED).end();
  }
}

export function tieneNivel(minimo) {
  return (req, res, next) => {
    if (req.user && req.user.nivel >= minimo) {
      next();
    } else {
      res.status(UNAUTHORIZED).end();
    }
  };
}
