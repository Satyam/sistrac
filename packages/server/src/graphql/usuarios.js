import { readUsuarios, readUsuarioPorUsuario } from '../dbOps/usuarios';

export const typeDefs = `

type Usuario {
  idUsuario: ID!
  password: String
  usuario: String
  nivel: Int
  rolGuarda: Boolean
  rolDios: Boolean
  rolSupervisor: Boolean
  rolMecanico: Boolean
  funcion: Int
  nombre: String
}

# This type specifies the entry points into our API.
extend type Query {
   usuarios: [Usuario]
   usuario(usuario: String, idUsuario: ID): Usuario
}
`;
export const resolvers = {
  Query: {
    usuarios: () => readUsuarios(),
    usuario: (parent, { usuario, idUsuario }) => {
      if (usuario) return readUsuarioPorUsuario(usuario);
      if (idUsuario) return readUsuario(parseInt(idUsuario, 10));
    },
  },
};
