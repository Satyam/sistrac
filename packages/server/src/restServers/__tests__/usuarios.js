import md5 from 'md5';
import jwt from 'jsonwebtoken';

import { SECRET, COOKIE_NAME } from '../../../config';

import {
  initRoutes,
  testREST,
  Response,
  GET,
  POST,
  PUT,
  DELETE,
} from '../../testUtils';

import {
  OK,
  UNAUTHORIZED,
  BAD_REQUEST,
  CONFLICT,
  NO_CONTENT,
  NOT_FOUND,
  CREATED,
} from '../httpStatusCodes';

import { authenticate } from '../usuarios';

const NUEVO_USUARIO = {
  usuario: 'pepe',
  password: '123456789',
  nombre: 'José Pérez',
};

const roles = {
  rolDios: false,
  rolGuarda: false,
  rolMecanico: false,
  rolSupervisor: false,
};

describe('restServers/usuarios', () => {
  beforeAll(() => {
    initRoutes();
  });
  describe('login', () => {
    it(
      'success',
      testREST({
        method: PUT,
        path: '/usuarios/login',
        queryResult: [
          { ...NUEVO_USUARIO, password: md5(NUEVO_USUARIO.password) },
        ],
        statusCode: OK,
        restResult: { ...NUEVO_USUARIO, ...roles, password: undefined },
        body: NUEVO_USUARIO,
      }),
    );
    it(
      'fail password',
      testREST({
        method: PUT,
        path: '/usuarios/login',
        queryResult: [
          { ...NUEVO_USUARIO, password: md5(NUEVO_USUARIO.password) },
        ],
        statusCode: UNAUTHORIZED,
        body: { ...NUEVO_USUARIO, password: 0 },
      }),
    );
    it(
      'fail usuario',
      testREST({
        method: PUT,
        path: '/usuarios/login',
        queryResult: [],
        statusCode: UNAUTHORIZED,
        body: { ...NUEVO_USUARIO, password: 0 },
      }),
    );
  });
  describe('signup', () => {
    it(
      'success',
      testREST({
        method: POST,
        path: '/usuarios/signup',
        queryResult: { insertId: 25 },
        statusCode: CREATED,
        body: NUEVO_USUARIO,
      }),
    );
    it(
      'fail faltan datos',
      testREST({
        method: POST,
        path: '/usuarios/signup',
        statusCode: BAD_REQUEST,
        restResult:
          'Favor de indicar código de usuario, contraseña y nombre completo.',
        body: { ...NUEVO_USUARIO, password: 0 },
      }),
    );
    it(
      'fail nombre existe',
      testREST({
        method: POST,
        path: '/usuarios/signup',
        queryResult: { affectedRows: 0 },
        statusCode: CONFLICT,
        body: NUEVO_USUARIO,
      }),
    );
  });
  describe('logout', () => {
    it(
      'success',
      testREST({
        method: GET,
        path: '/usuarios/logout',
        statusCode: NO_CONTENT,
      }),
    );
  });
  describe('userData', () => {
    it(
      'success',
      testREST({
        method: GET,
        path: '/usuarios/pepe',
        queryResult: [NUEVO_USUARIO],
        statusCode: OK,
        restResult: { ...NUEVO_USUARIO, ...roles, password: undefined },
      }),
    );
    it(
      'fail',
      testREST({
        method: GET,
        path: '/usuarios/pepe',
        queryResult: [],
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('delete', () => {
    it(
      'success',
      testREST({
        method: DELETE,
        path: '/usuarios/25',
        queryResult: { affectedRows: 1 },
        statusCode: NO_CONTENT,
      }),
    );
    it(
      'fail',
      testREST({
        method: DELETE,
        path: '/usuarios/999',
        queryResult: { affectedRows: 0 },
        statusCode: NOT_FOUND,
      }),
    );
  });
  describe('update', () => {
    it(
      'success',
      testREST({
        method: PUT,
        path: '/usuarios/25',
        queryResult: { affectedRows: 1 },
        statusCode: NO_CONTENT,
        body: NUEVO_USUARIO,
      }),
    );
    it(
      'fail',
      testREST({
        method: PUT,
        path: '/usuarios/999',
        queryResult: { affectedRows: 0 },
        statusCode: NOT_FOUND,
        body: NUEVO_USUARIO,
      }),
    );
  });
  describe('authenticate', () => {
    it('success', done => {
      const req = {
        method: GET,
        path: 'whatever',
        cookies: {
          [COOKIE_NAME]: jwt.sign({ ...NUEVO_USUARIO, ...roles }, SECRET),
        },
      };

      authenticate(
        req,
        new Response(() => {
          expect('Should not come this way').toBeFalsy();
        }),
        () => {
          expect(req.user).toEqual({ ...NUEVO_USUARIO, ...roles });
          done();
        },
      );
    });
    it('skip on login', done => {
      const req = {
        method: GET,
        path: '.... /login',
        cookies: {
          [COOKIE_NAME]: jwt.sign({ ...NUEVO_USUARIO, ...roles }, SECRET),
        },
      };
      authenticate(
        req,
        new Response(() => {
          expect('Should not come this way').toBeFalsy();
        }),
        () => {
          expect(req.user).toBeUndefined();
          done();
        },
      );
    });
    it('skip on signup', done => {
      const req = {
        method: GET,
        path: '.... /login',
        cookies: {
          [COOKIE_NAME]: jwt.sign({ ...NUEVO_USUARIO, ...roles }, SECRET),
        },
      };
      authenticate(
        req,
        new Response(() => {
          expect('Should not come this way').toBeFalsy();
        }),
        () => {
          expect(req.user).toBeUndefined();
          done();
        },
      );
    });
    it('fail bad token', done => {
      const req = {
        method: GET,
        path: 'whatever',
        cookies: {
          [COOKIE_NAME]: jwt.sign(
            { ...NUEVO_USUARIO, ...roles },
            `${SECRET} asdf`,
          ),
        },
      };

      authenticate(
        req,
        new Response(status => {
          expect(status).toBe(UNAUTHORIZED);
          expect(req.user).toBeUndefined();
          done();
        }),
        () => {
          expect('Should not come this way').toBeFalsy();
        },
      );
    });
    it('fail no token', done => {
      const req = {
        method: GET,
        path: 'whatever',
        cookies: {},
      };
      authenticate(
        req,
        new Response(status => {
          expect(status).toBe(UNAUTHORIZED);
          expect(req.user).toBeUndefined();
          done();
        }),
        () => {
          expect('Should not come this way').toBeFalsy();
        },
      );
    });
  });
});
