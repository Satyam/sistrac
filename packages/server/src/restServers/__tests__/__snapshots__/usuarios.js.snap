// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restServers/usuarios delete fail 1`] = `Object {}`;

exports[`restServers/usuarios delete fail 2`] = `"delete from Usuarios where idUsuario = 999"`;

exports[`restServers/usuarios delete success 1`] = `Object {}`;

exports[`restServers/usuarios delete success 2`] = `"delete from Usuarios where idUsuario = 25"`;

exports[`restServers/usuarios login fail password 1`] = `Object {}`;

exports[`restServers/usuarios login fail password 2`] = `"select * from Usuarios where usuario = 'pepe'"`;

exports[`restServers/usuarios login fail usuario 1`] = `Object {}`;

exports[`restServers/usuarios login fail usuario 2`] = `"select * from Usuarios where usuario = 'pepe'"`;

exports[`restServers/usuarios login success 1`] = `
Object {
  "user": Object {
    "nombre": "José Pérez",
    "rolDios": false,
    "rolGuarda": false,
    "rolMecanico": false,
    "rolSupervisor": false,
    "usuario": "pepe",
  },
}
`;

exports[`restServers/usuarios login success 2`] = `"select * from Usuarios where usuario = 'pepe'"`;

exports[`restServers/usuarios logout success 1`] = `Object {}`;

exports[`restServers/usuarios logout success 2`] = `undefined`;

exports[`restServers/usuarios signup fail faltan datos 1`] = `Object {}`;

exports[`restServers/usuarios signup fail faltan datos 2`] = `undefined`;

exports[`restServers/usuarios signup fail nombre existe 1`] = `Object {}`;

exports[`restServers/usuarios signup fail nombre existe 2`] = `"insert into Usuarios set \`usuario\` = 'pepe', \`password\` = '25f9e794323b453885f5181f1b624d0b', \`nombre\` = 'José Pérez'"`;

exports[`restServers/usuarios signup success 1`] = `Object {}`;

exports[`restServers/usuarios signup success 2`] = `"insert into Usuarios set \`usuario\` = 'pepe', \`password\` = '25f9e794323b453885f5181f1b624d0b', \`nombre\` = 'José Pérez'"`;

exports[`restServers/usuarios update fail 1`] = `Object {}`;

exports[`restServers/usuarios update fail 2`] = `"update Usuarios set \`usuario\` = 'pepe', \`password\` = '123456789', \`nombre\` = 'José Pérez' where idUsuario = 999"`;

exports[`restServers/usuarios update success 1`] = `Object {}`;

exports[`restServers/usuarios update success 2`] = `"update Usuarios set \`usuario\` = 'pepe', \`password\` = '123456789', \`nombre\` = 'José Pérez' where idUsuario = 25"`;

exports[`restServers/usuarios userData fail 1`] = `Object {}`;

exports[`restServers/usuarios userData fail 2`] = `"select * from Usuarios where usuario = 'pepe'"`;

exports[`restServers/usuarios userData success 1`] = `Object {}`;

exports[`restServers/usuarios userData success 2`] = `"select * from Usuarios where usuario = 'pepe'"`;
